### PR TITLE
dependency updates 20231020

### PR DESCRIPTION
**Pull request by lrworth**
- flake.lock: Update
- chore: add more tested-by

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205728013439142

### CODE REVIEW

File: .github/workflows/haskell-ci.yml
Comment (JackKelly-Bellroy) [inline .github/workflows/haskell-ci.yml:1]: Could mark these files as generated if we wanted. But I also like the diff

File: wai-handler-hal.cabal [hunk 1]
```diff
@@ -28,7 +28,13 @@ extra-source-files:
   test/golden/WaiRequest.txt
 
 tested-with:
-  GHC ==8.6.5 || ==8.8.4 || ==8.10.7 || ==9.0.2 || ==9.2.2
+  GHC ==8.6.5
+   || ==8.8.4
```
Comment (JackKelly-Bellroy) [inline wai-handler-hal.cabal:32]: Pretty sure the nixpkgs doesn't go back to 8.6.5 any more, which is why I cut `aws-arn` back to `8.10.7`. Up to you

**Review by JackKelly-Bellroy — APPROVED** [review]
Assuming CI passes, revise hackage and merge.

**Comment by lrworth** [comment]
https://hackage.haskell.org/package/wai-handler-hal-0.2.0.0/revisions/

File: wai-handler-hal.cabal (hunk 1 above)
Comment (lrworth) [reply wai-handler-hal.cabal:32]: We have CI testing it so may as well keep it until that goes away.

**Pull request merged**